digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectValidatorSpec.scala

```diff
@@ -1899,6 +1899,58 @@ Left|      |Right
     }
   }
 
+  test("Test checkProjectElyCodes When discontinuity is 3 on one track but not on the other of a road part Then validator should return errors") {
+    runWithRollback {
+      val raId1 = Sequences.nextRoadwayId
+      val raId2 = Sequences.nextRoadwayId
+      val startDate = DateTime.now()
+      val linearLocationId1 = Sequences.nextLinearLocationId
+      val linearLocationId2 = Sequences.nextLinearLocationId
+      val ra = Seq(
+        RoadAddress(12345, 1, 16320L, 2L, AdministrativeClass.State, Track.LeftSide, Discontinuity.EndOfRoad, 0, 10, Some(DateTime.parse("1901-01-01")), None, Some("User"), 1000, 0, 10, AgainstDigitizing, DateTime.now().getMillis, (None, None), Seq(Point(0.0, 40.0), Point(0.0, 50.0)), LinkGeomSource.NormalLinkInterface, 8, NoTermination, roadwayNumber1, Some(DateTime.parse("1901-01-01")), None, None),
+        RoadAddress(12346, 1, 16320L, 2L, AdministrativeClass.State, Track.RightSide, Discontinuity.EndOfRoad, 0, 10, Some(DateTime.parse("1901-01-01")), None, Some("User"), 1000, 0, 10, AgainstDigitizing, DateTime.now().getMillis, (None, None), Seq(Point(0.0, 40.0), Point(0.0, 50.0)), LinkGeomSource.NormalLinkInterface, 8, NoTermination, roadwayNumber1, Some(DateTime.parse("1901-01-01")), None, None)
+      )
+
+      val roadway1 = Roadway(raId1, roadwayNumber1, 16320L, 2L, AdministrativeClass.State, Track.LeftSide, Discontinuity.Continuous, 0L, 10L, reversed = false, DateTime.now(), None, "test_user", None, 8, NoTermination, startDate, None)
```

Comment:
Itse testin kannalta ei taida olla väliä - mutta tota...
Eiks RoadAddressien pitäis olla yhdistelmä Roadwayistä ja Linear locationeista

Esim. jatkuvuuskoodit ovat nyt RoadAddresseissa EndOfRoad, ja roadwayissa Continuous, ja digitointisuunnat toisissa against ja toisissa towards. Pitääkssenmukaollannäin, onksnoioikein?

Nyt jäin pähkimään, oliks noilla eroilla joku merkitys johonkin testin kohtaan. (Tuskin, mutta hämmentää lukijaa)

(ra: yhdistelmä ll+rw-> sen puoleen, järjestyskin olisi järkevämpi, jos ll:t ja roadwayt olisivat ra:ita ennen)